src/datastore/postgres-store.ts

```diff
@@ -2158,6 +2163,35 @@ export class PgDataStore
     return { removedTxs: removedTxs };
   }
 
+  /**
+   * Deletes mempool txs older than `STACKS_MEMPOOL_TX_GARBAGE_COLLECTION_THRESHOLD` blocks (default 256).
+   * @param client - DB client
+   * @returns List of deleted `tx_id`s
+   */
+  async deleteGarbageCollectedMempoolTxs(client: ClientBase): Promise<{ deletedTxs: string[] }> {
+    // Get threshold block.
+    const blockThreshold = process.env['STACKS_MEMPOOL_TX_GARBAGE_COLLECTION_THRESHOLD'] ?? 256;
+    const cutoffResults = await client.query<{ block_height: number }>(
+      `SELECT (block_height - $1) AS block_height FROM chain_tip`,
+      [blockThreshold]
+    );
+    if (cutoffResults.rowCount != 1) {
+      return { deletedTxs: [] };
+    }
+    const cutoffBlockHeight = cutoffResults.rows[0].block_height;
+    // Delete every mempool tx that came before that block.
+    // TODO: Use DELETE instead of UPDATE once we implement a non-archival API replay mode.
+    const deletedTxResults = await client.query<{ tx_id: Buffer }>(
+      `UPDATE mempool_txs
+      SET pruned = TRUE, status = $2
+      WHERE receipt_block_height < $1
```

Comment:
Should this `where` condition exclude rows that were already pruned? Otherwise I think it would overwrite existing `status` columns, right?

Reply:
Oh, great catch. You're right.